test/v1/LPNRegistryV1.t.sol

```diff
@@ -0,0 +1,235 @@
+            TEST_START_BLOCK,
+            TEST_END_BLOCK,
+            FEE,
+            0 // proofBlock is 0 for L1
+        );
+        // do the thing
+        uint256 id = registry.request{value: FEE}(
+            QUERY_HASH, PLACEHOLDERS, TEST_START_BLOCK, TEST_END_BLOCK
+        );
+        // check request data
+        (address client, QueryInput memory input) = registry.requests(id);
+        assertEq(client, stranger);
+        assertEq(input.limit, 0);
+        assertEq(input.offset, 0);
+        assertEq(input.minBlockNumber, TEST_START_BLOCK);
+        assertEq(input.maxBlockNumber, TEST_END_BLOCK);
+        assertEq(input.blockHash, 0); // TODO - only on ethereum
+        assertEq(input.computationalHash, QUERY_HASH);
+        assertEq(
+            keccak256(abi.encode(input.userPlaceholders)),
+            keccak256(abi.encode(PLACEHOLDERS))
+        );
+        // make duplicate request
+        uint256 id2 = registry.request{value: FEE}(
+            QUERY_HASH, PLACEHOLDERS, TEST_START_BLOCK, TEST_END_BLOCK
+        );
+        assertNotEq(id, id2);
+    }
+
+    function test_request_insufficientGasFee_reverts() public {
+        vm.prank(stranger);
+        vm.expectRevert(QueryManager.InsufficientGasFee.selector);
+        registry.request{value: 0}(
+            QUERY_HASH, PLACEHOLDERS, TEST_START_BLOCK, TEST_END_BLOCK
+        );
+    }
+
+    function test_request_invalidRange_reverts() public {
+        vm.prank(stranger);
+        vm.expectRevert(QueryManager.QueryInvalidRange.selector);
+        registry.request{value: FEE}(
+            QUERY_HASH, PLACEHOLDERS, TEST_END_BLOCK, TEST_START_BLOCK
+        );
+    }
+
+    function test_request_exceedsMaxRange_reverts() public {
+        vm.skip(true); // TODO - maxRange not enforced correctly
+        uint256 maxRange = registry.MAX_QUERY_RANGE();
+        vm.startPrank(stranger);
+        // max range should succeed
+        registry.request{value: FEE}(
+            QUERY_HASH,
+            PLACEHOLDERS,
+            TEST_START_BLOCK,
+            TEST_START_BLOCK + maxRange
+        );
+        // max range + 1 should revert
+        vm.expectRevert(QueryManager.QueryGreaterThanMaxRange.selector);
+        registry.request{value: FEE}(
+            QUERY_HASH,
+            PLACEHOLDERS,
+            TEST_START_BLOCK,
+            TEST_START_BLOCK + maxRange + 1
+        );
+    }
+
+    function test_request_afterCurrentBlock_reverts() public {
+        uint256 futureBlock = block.number + 1;
+        vm.prank(stranger);
+        vm.expectRevert(QueryManager.QueryAfterCurrentBlock.selector);
+        registry.request{value: FEE}(
+            QUERY_HASH, PLACEHOLDERS, TEST_START_BLOCK, futureBlock
+        );
+    }
+
+    function test_respond_success() public {
+        // to properly test this we need to either:
+        //   a. mock the groth16 verifier library or ...
+        //   b. generate a real, working proof for this test (preferred)
```

Comment:
We can give you a real working proof but it wont necessarily follow the latest params onchain. If we could manage to have a test version of Groth16Verifier that is fixed (it only contains the cryptographic verification, no lagrange specific logic) then we could do it.

Reply:
yeah I noticed there is a proof sitting in `test/v0/full_proof.bin`, but I imagine it is old and no longer functional. We need a way to test this without having to manually update a binary every time the groth verifier changes 🤔